### ISSUE TITLE
Top-level comments in extending templates

### ISSUE THREAD

**Issue by tomkersten** [pugjs/pug#2632]
I see that comments are no longer allowed at the top-level of extending templates. I can work on a PR for this if desired but wanted to confirm whether it was an accidental or intentional change.

```bash
Message:
    /.../file.pug:3

Only named blocks and mixins can appear at the top level of an extending template
Details:
    code: PUG:UNEXPECTED_NODES_IN_EXTENDING_ROOT
    msg: Only named blocks and mixins can appear at the top level of an extending template
    line: 3
    column: undefined
    filename: /.../file.pug
    src: undefined
    toJSON: function () {
    return {
      code: this.code,
      msg: this.msg,
      line: this.line,
      column: this.column,
      filename: this.filename
    };
  }
```

Thoughts?

**Comment by franciscop** [comment]
I am also suffering from this when migrating from `jade` to `pug`

**Comment by tomkersten** [comment]
It's not a _huge_ deal to me, but, it is nice to have comments to document templates/usage (what are available blocks, et cetera)...and it seemed like it may have just been an accidental oversight.

**Comment by franciscop** [comment]
@tomkersten sorry for not explaining properly, **you can have pug comments** but you cannot have HTML comments. Just add a `-` after your `//` so it looks like `//- my comment`.

**Comment by tomkersten** [comment]
Ah. Nice. I'll give that a try.

**Comment by ForbesLindesay** [comment]
Yes, the key here is that comments like `// a comment` are compiled to html like `<!--a comment-->` so it kind of makes sense for pug to reject them in places where we cannot render the output.  However comment like `//- a comment` (this is called an "unbuffered comment) do not produce any output, so they can go anywhere.

I would, however, be happy to accept a pull request that ignores comments at the top level of a file that is extending another template.  The code would, I believe, go here: https://github.com/pugjs/pug-linker/blob/master/index.js#L34
